ios/brave-ios/Sources/PlaylistUI/MediaScrubber.swift

```diff
@@ -0,0 +1,246 @@
+      isScrubbing = newValue
+    }
+    .accessibilityRepresentation {
+      if duration > 0 {
+        Slider(
+          value: $currentTime,
+          in: 0.0...duration,
+          step: 1
+        ) {
+          Text("Current Media Time")  // TODO: Localize
+        } minimumValueLabel: {
+          currentValueLabel
+        } maximumValueLabel: {
+          durationLabel
+        }
+      }
+    }
+  }
+}
+
+@available(iOS 16.0, *)
+extension MediaScrubber where Label == DefaultMediaScrubberLabel {
+  init(
+    currentTime: Binding<TimeInterval>,
+    duration: TimeInterval,
+    isScrubbing: Binding<Bool>
+  ) {
+    self._currentTime = currentTime
+    self.duration = duration
+    self._isScrubbing = isScrubbing
+    self.label = DefaultMediaScrubberLabel(
+      currentTime: currentTime.wrappedValue,
+      duration: duration
+    )
+  }
+}
+
+@available(iOS 16.0, *)
+struct DefaultMediaScrubberLabel: View {
+  var currentTime: TimeInterval
+  var duration: TimeInterval
+
+  @State private var isShowingTotalTime: Bool = false
+
+  private var currentValueLabel: Text {
+    return Text(.seconds(currentTime), format: .time(pattern: .minuteSecond))
+  }
+
+  private var remainingTimeLabel: Text {
+    return Text(.seconds(currentTime - duration), format: .time(pattern: .minuteSecond))
+  }
+
+  private var durationLabel: Text {
+    Text(.seconds(duration), format: .time(pattern: .minuteSecond))
+  }
+
+  var body: some View {
+    HStack {
+      currentValueLabel
+      Spacer()
+      Button {
+        isShowingTotalTime.toggle()
+      } label: {
+        Group {
+          if isShowingTotalTime {
+            durationLabel
+          } else {
+            remainingTimeLabel
+          }
+        }
+        .transition(.move(edge: .trailing).combined(with: .opacity))
+      }
+    }
+    .foregroundStyle(.primary)
+    .font(.footnote)
+  }
+}
+
+@available(iOS 16.0, *)
+extension Duration {
+  fileprivate var isValid: Bool {
+    components.seconds > 0
+  }
+  fileprivate var seconds: Double {
+    Double(components.seconds)
+  }
+}
+
+#if DEBUG
+@available(iOS 16.0, *)
+private struct MediaScrubberPreview: View {
+  @State private var currentTime: TimeInterval = 0
+  @State private var isScrubbing: Bool = false
+
+  var body: some View {
+    VStack {
+      MediaScrubber(
+        currentTime: $currentTime,
+        duration: 1000,
+        isScrubbing: $isScrubbing
+      )
+      .padding()
+      MediaScrubber(
+        currentTime: $currentTime,
+        duration: 1000,
+        isScrubbing: $isScrubbing
+      )
+      .tint(.red)
+      .environment(\.colorScheme, .dark)
+      .padding()
+      .background(Color.black)
+
+      Button {
+        // FIXME: Currently animation is linear(0.1) based on animations in the actual MediaScrubber, see if its possible to only use those animations while scrubbing
+        //        withAnimation(.spring()) {
+        currentTime = 500
```

Comment:
Why 500 for Interval ?

Reply:
This is a SwiftUI preview, so its just random